### PR TITLE
feat(controllers): add INTEGRATION* gateway events

### PR DESCRIPTION
**Pull request by ayntee**
# THIS PR WILL REQUIRE PROPER REVIEW

Closes: #457 
Reference: https://github.com/discord/discord-api-docs/pull/2533

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -7,6 +7,6 @@ jobs:
       - uses: actions/checkout@v2
       - uses: denolib/setup-deno@v2
       - name: Run fmt check script
-        run: deno fmt --check
+        run: deno fmt --check --ignore=./src/types/util.ts
```
Comment (ayntee) [inline .github/workflows/lint.yml:10]: `deno fmt` is bugged

File: src/api/controllers/misc.ts [hunk 2]
```diff
@@ -155,3 +157,78 @@ export function handleInternalWebhooksUpdate(data: DiscordPayload) {
...
+    ...rest,
+    applicationID,
+    guildID,
+  });
+}
+
+export type IntegrationCreateUpdateEvent = Integration & {
+  /** id of the guild */
+  guild_id: string;
+};
```
Comment (itohatweb) [inline src/api/controllers/misc.ts:234]: Please move this to `types/discord.ts`

File: src/types/options.ts [hunk 3]
```diff
@@ -208,6 +210,21 @@ export interface EventHandlers {
...
+}
+
+export interface IntegrationDeleteEvent {
+  /** integration id */
+  id: string;
+  /** id of the guild */
+  guild_id: string;
+  /** id of the bot/OAuth2 application for this discord integration */
+  application_id?: string;
 }
```
Comment (itohatweb) [inline src/types/options.ts:228]: This should be in `types/discord.ts`

**Comment by itohatweb** [comment]
image

**Comment by itohatweb** [comment]
Please add the added integration events to the `mod.ts` file

File: .github/workflows/lint.yml (hunk 1 above)
Comment (itohatweb) [reply .github/workflows/lint.yml:10]: I don't think this should be needed.

File: src/types/util.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export type CamelizeString<T extends PropertyKey> = T extends string
```
Comment (itohatweb) [inline src/types/util.ts:1]: consider moving that to `./src/util/utils.ts`

Comment (ayntee) [reply src/types/util.ts:1]: No

File: src/api/controllers/misc.ts [hunk 5]
```diff
@@ -155,3 +157,73 @@ export function handleInternalWebhooksUpdate(data: DiscordPayload) {
     options.guild_id,
   );
 }
+
+export function handleInternalIntegrationCreate(
+  data: DiscordPayload,
+) {
+  if (data.t !== "INTEGRATION_CREATE") return;
+
+  const {
```
Comment (Skillz4Killz) [inline src/api/controllers/misc.ts:166]: Use the snake case stuff for these now

**Review by Skillz4Killz — CHANGES_REQUESTED** [review]

File: src/api/controllers/misc.ts (hunk 5 above)
Comment (ayntee) [reply src/api/controllers/misc.ts:166]: It should be fine to use camel case for the data supplied to the user for now, once the "snake case to camel case" PR is merged, we can use it to convert the properties into camel case easily.

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**